pkg/cloudprovider/aws/instancetype.go

```diff
@@ -319,19 +312,65 @@ func (i *InstanceType) kubeReservedResources(pods int64) v1.ResourceList {
 			resources[v1.ResourceCPU] = *cpuOverhead
 		}
 	}
+	if kc != nil && kc.KubeReserved != nil {
+		return lo.Assign(resources, kc.KubeReserved)
+	}
 	return resources
 }
 
-func (i *InstanceType) miscResources(vmMemOverhead float64) v1.ResourceList {
+func (i *InstanceType) evictionThreshold(kc *v1alpha5.KubeletConfiguration, vmMemoryOverhead resource.Quantity) v1.ResourceList {
+	overhead := v1.ResourceList{
+		v1.ResourceMemory: resource.MustParse("100Mi"),
+	}
+	if kc != nil && kc.EvictionHard != nil {
```

Comment:
might be better in this case to invert this logic and return overhead early rather than doing 4-nested ifs